File: tests/if.swift
var a = 10;
var result:Int = 0;
if (a > 10) {
  result = 0;
} else if (a > 100) {
  result = 1;
} else {
  result = 0;
}

expect(result == 0);

if (result == 0) {
  result = 1337;
}

expect(result == 1337);

if (result == 1337) {
  result = 1338;
} else {
  result = 1336;
}

expect(result == 1338);

File: src/Parser/branch/if.js
import {
  Token,
  Types as Type,
  TokenList as TT
} from "../../labels";

import Node from "../../nodes";

/**
 * @return {Node}
 */
export function parseIfStatement() {

  let node = new Node.IfStatement();

  if (this.eat(TT.IF)) {
    node.test = this.parseCondition();
  }

  // Consequent
  this.expect(TT.LBRACE);
  node.consequent = this.parseBlock();
  this.expect(TT.RBRACE);

  // Alternate: else|else if
  if (this.eat(TT.ELSE)) {
    node.alternate = this.parseIfStatement();
  }

  return (node);

}